lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -590,8 +590,10 @@ Distribution TruncatedDistribution::getMarginal(const Indices & indices) const
   if (useSimplifiedVersion_)
     return simplifiedVersion_.getMarginal(indices);
 
-  Interval marginalBounds(bounds_.getMarginal(indices));
-  return new TruncatedDistribution(distribution_.getMarginal(indices), marginalBounds);
+  if (distribution_.hasIndependentCopula())
```

Comment:
This case is already taken into account in the simplification mechanism

Reply:
there is no simplification for general n-d distribution